dev/bots/prepare_package.dart

```diff
@@ -899,16 +911,19 @@ Future<void> main(List<String> rawArguments) async {
   try {
     final Map<String, String> version = await creator.initializeRepo();
     final File outputFile = await creator.createArchive();
+    final ArchivePublisher publisher = ArchivePublisher(
+      tempDir,
+      revision,
+      branch,
+      version,
+      outputFile,
+      dryRun,
+    );
     if (parsedArguments['publish'] as bool) {
-      final ArchivePublisher publisher = ArchivePublisher(
-        tempDir,
-        revision,
-        branch,
-        version,
-        outputFile,
-        dryRun,
-      );
       await publisher.publishArchive(parsedArguments['force'] as bool);
```

Comment:
I think this flow would be easier to read if you factored out the logic to generate the metadata out of `publishArchive()`, and then always call `generateLocalMetadata()` and then only conditionally call `publisher.publishArchive(...)` if the flag is provided.

Reply:
Done.